kyo-core/shared/src/main/scala/kyo/Log.scala

```diff
@@ -2,7 +2,7 @@ package kyo
 
 import kyo.internal.LogPlatformSpecific
 
-final case class Log(unsafe: Log.Unsafe) extends AnyVal:
+final case class Log(unsafe: Log.Unsafe):
```

Comment:
I noticed several `Log` allocations in the benchmark. There are scenarios where both the compiler and the JIT fail to prevent allocation, triggering the boxing every time `Log` is used. By removing `AnyVal`, the `Log` is always allocated once when created and then reused. I'll take a look at making this change for the other effects as well.